app/src/main/java/com/github/se/polyfit/data/processor/LocalDataProcessor.kt

```diff
@@ -15,46 +16,66 @@ data class DailyCalorieSummary(val date: LocalDate, val totalCalories: Double)
+   */
   fun calculateCaloriesSince(sinceDate: LocalDate): List<DailyCalorieSummary> {
-    return mealDao
-        .getMealsCreatedOnOrAfterDate(sinceDate, user.id)
-        .filter { it.isComplete() && it.occasion != MealOccasion.OTHER }
-        .groupBy { it.createdAt }
-        .map { (date, meals) ->
-          DailyCalorieSummary(date, meals.sumOf { it.calculateTotalCalories() })
-        }
+    return try {
+      mealDao
+          .getMealsCreatedOnOrAfterDate(sinceDate, user.id)
+          .filter { it.isComplete() && it.occasion != MealOccasion.OTHER }
+          .groupBy { it.createdAt }
+          .map { (date, meals) ->
+            DailyCalorieSummary(date, meals.sumOf { it.calculateTotalCalories() })
+          }
+    } catch (e: Exception) {
+      Log.e("LocalDataProcessor", "Error getting calories since $sinceDate", e)
+      emptyList()
+    }
   }
 
   fun getCaloriesPerMealOccasionToday(): Map<MealOccasion, Double> {
     return getCaloriesPerMealOnSpecificDay(LocalDate.now())
   }
 
+  /**
+   * Get the calories per meal occasion today as a LiveData object. If an error occurs, an empty
+   * list is returned.
+   */
   fun getCaloriesPerMealOccasionTodayLiveData(): LiveData<List<Pair<MealOccasion, Double>>> {
-    Log.d("LocalDataProcessor", "${LocalDate.now()}")
-    return mealDao.getMealsCreatedOnDateLiveData(LocalDate.now(), user.id).map { meals ->
-      val allOccasion =
-          MealOccasion.values().toList().filter { it != MealOccasion.OTHER }.map { Pair(it, 0.0) }
-
-      val mealValues =
-          meals
-              .toMutableList()
-              .filter { it.isComplete() && it.occasion != MealOccasion.OTHER }
-              .groupBy { it.occasion }
-              .mapValues { (_, meals) -> meals.sumOf { it.calculateTotalCalories() } }
-              .toList()
+    return try {
+      mealDao.getMealsCreatedOnDateLiveData(LocalDate.now(), user.id).map { meals ->
+        val allOccasion =
+            MealOccasion.entries.filter { it != MealOccasion.OTHER }.map { Pair(it, 0.0) }
 
-      Log.d("LocalDataProcessor", "mealValues: $mealValues")
+        val mealValues =
+            meals
+                .toMutableList()
+                .filter { it.isComplete() && it.occasion != MealOccasion.OTHER }
+                .groupBy { it.occasion }
+                .mapValues { (_, meals) -> meals.sumOf { it.calculateTotalCalories() } }
+                .toList()
 
-      allOccasion.map {
-        val (occasion, _) = it
-        val value = mealValues.find { it.first == occasion }?.second ?: 0.0
-        Pair(occasion, value)
+        allOccasion.map { pair ->
+          val (occasion, _) = pair
+          val value = mealValues.find { it.first == occasion }?.second ?: 0.0
+          Pair(occasion, value)
+        }
       }
+    } catch (e: Exception) {
+      Log.e("LocalDataProcessor", "Error getting calories per meal occasion today", e)
+      MutableLiveData(emptyList())
     }
   }
```

Comment:
Might be a good idea to create something like 
`val result = MutableLiveData<List<Pair<MealOccasion, Double>>>()`
at the beginning of the function and make a `result.postValue(...)` to set its value inside your try catch block.

Reply:
That's a great idea! I tried implementing it. Here's what I came up with:

```kotlin
fun getCaloriesPerMealOccasionTodayLiveData(): LiveData<List<Pair<MealOccasion, Double>>> {
    val result = MutableLiveData<List<Pair<MealOccasion, Double>>>()
    
    try {
        val liveData = mealDao.getMealsCreatedOnDateLiveData(LocalDate.now(), user.id)
        liveData.map { meals ->
            val allOccasion =
                MealOccasion.entries.filter { it != MealOccasion.OTHER }.map { Pair(it, 0.0) }

            val mealValues =
                meals
                    .filter { it.isComplete() && it.occasion != MealOccasion.OTHER }
                    .groupBy { it.occasion }
                    .mapValues { (_, meals) -> meals.sumOf { it.calculateTotalCalories() } }
                    .toList()

            allOccasion.map { pair ->
                val (occasion, _) = pair
                val value = mealValues.find { it.first == occasion }?.second ?: 0.0
                Pair(occasion, value)
            }
        }.observeForever { result.postValue(it) }
    } catch (e: Exception) {
        Log.e("LocalDataProcessor", "Error getting calories per meal occasion today", e)
        result.postValue(emptyList())
    }

    return result
}

``` 

The issue with that is the `observeForever` needed in this case, which can potentially cause memory leaks and degrade performance.